File: src/main/java/uk/co/eelpieconsulting/osm/nominatim/controllers/ImportController.java
package uk.co.eelpieconsulting.osm.nominatim.controllers;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.servlet.ModelAndView;
import uk.co.eelpieconsulting.osm.nominatim.indexing.FullIndexBuilder;
import uk.co.eelpieconsulting.osm.nominatim.views.ViewFactory;

import java.io.FileNotFoundException;
import java.io.IOException;
import java.sql.SQLException;

@Controller
public class ImportController {
	
	private final FullIndexBuilder fullIndexBuilder;
	private final ViewFactory viewFactory;
	
	@Autowired
	public ImportController(FullIndexBuilder fullIndexBuilder, ViewFactory viewFactory) {
		this.fullIndexBuilder = fullIndexBuilder;
		this.viewFactory = viewFactory;
	}
	
	@RequestMapping("/import")
	public ModelAndView inputIndex() throws FileNotFoundException, IOException, SQLException {
		try {
			fullIndexBuilder.buildFullIndex();
			return new ModelAndView(viewFactory.getJsonView()).addObject("data", "ok");
		} catch(Exception e) {
			e.printStackTrace();
			return new ModelAndView(viewFactory.getJsonView()).addObject("data", "error: " + e.getMessage());
		}
	}
	
}

File: src/main/kotlin/uk/co/eelpieconsulting/osm/nominatim/controller/ImportController.kt
package uk.co.eelpieconsulting.osm.nominatim.controller

import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RestController
import uk.co.eelpieconsulting.osm.nominatim.indexing.FullIndexBuilder

@RestController
class ImportController(val fullIndexBuilder: FullIndexBuilder) {

    @PostMapping("/import")
    fun inputIndex(): String {
        fullIndexBuilder.buildFullIndex()
        return "ok"
    }

}
